jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/applications/AliasEndpoint.java

```diff
@@ -225,14 +265,17 @@ public Response updateAliasPassword(@PathParam("alias") String alias, PasswordCo
 			@Context HttpServletRequest request) {
 
 		// Endpoint is not protected by keycloak, we assume Bearer exists.
-		// TODO - Find a way to validate current user password.
 		try {
 			AccessToken accessToken = BearerHelper.getTokenDataFromBearer(request).orNull();
 			if (accessToken != null && accessToken.getPreferredUsername().equals(alias)) {
 				ResponseBuilder response = Response.notModified();
 				if (passwordContainer.isDataValid()) {
-					aliasService.updateAliasePassword(alias, passwordContainer.getCurrentPassword(),
-							passwordContainer.getNewPassword());
+
+					String issuer = accessToken.getIssuer();
+					String jwtApplicationName = URLUtils.getLastPart(issuer);
+
```

Comment:
Please document what is done here. 
the name `lastPart` is raising my concern. note the in production there are client names such as `c-test-dashboard-demo` which translate to the subdomain  c-test-dashboard-demo-mcs.c-b4.com

Reply:
This is not taken from hostname, but from the issuer url.